pkg/btf/validation.go

```diff
@@ -325,10 +325,6 @@ func GetSyscallsList() ([]string, error) {
 	for _, key := range syscallinfo.SyscallsNames() {
 		var fn *btf.Func
 
-		if key == "" {
```

Comment:
Why is this check being removed?

Reply:
In the bug fixed here, we created a list with 300+ zeros and then append after them, only to skip the zero in the calling method.

Reply:
this is a workaround for the bug introduced that produces a slice two times "too big" with plenty of zeros

Reply:
I think I'd prefer (especially as it was catching a bug previously) that we keep the check but return an error instead of 'continue'. That way, if it ever happens then we should see it fail, instead of upsetting the rest of the loop.

Reply:
So basically the previous code was doing that, and doing "continue" on zero: https://go.dev/play/p/Ek0lSbDg3UQ.

The fix does it correctly: https://go.dev/play/p/aNxMhmxSnRQ.

Did you take a look at the `SyscallsNames()` function? There is no way for it to return an entry with an empty string, it was just put because the author did not understand why there was empty string because of the error in the creation of the slice. The `syscallNames` map does not contain an empty key, it would be a bug in the list.

```go
func SyscallsNames() []string {
	ret := make([]string, 0, len(syscallNames))

	for _, v := range syscallNames {
		ret = append(ret, v)
	}
	return ret
}
```

Reply:
> There is no way

***Today*** There is no way...

Reply:
Those are not even keys, those are the values that are misnamed, the map is int[string]